File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/mapping/entity/UserRoleMapping.java
package com.nivasafinance.features.rolemanagement.mapping.entity;

import com.nivasafinance.common.audit.AuditableEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "n_user_role_mapping")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class UserRoleMapping extends AuditableEntity {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "username", nullable = false)
    private String username;
    
    @Column(name = "role", nullable = false)
    private String role;
    
    @Column(name = "is_primary", columnDefinition = "boolean", nullable = false)
    private Boolean isPrimary;
}

File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/mapping/entity/UserRoleMapping.kt
package com.nivasafinance.features.rolemanagement.mapping.entity

import com.nivasafinance.common.audit.AuditableEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.util.UUID

@Entity
@Table(name = "user_role_mapping")
class UserRoleMapping(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "UUID")
    val id: UUID? = null,

    @Column(name = "username", nullable = false, unique = true)
    val username: String,

    @Column(name = "role", nullable = false)
    val role: String
) : AuditableEntity()
